packages/backend-common/src/database/sqlite3.ts

```diff
@@ -47,12 +63,33 @@ export function createSqliteDatabaseClient(
 export function buildSqliteDatabaseConfig(
   dbConfig: Config,
   overrides?: Knex.Config,
-) {
-  return mergeDatabaseConfig(
-    dbConfig.get(),
+): Knex.Config {
+  const baseConfig = dbConfig.get<Knex.Config>();
+
+  // Normalize config to always contain a connection object
+  if (typeof baseConfig.connection === 'string') {
+    baseConfig.connection = { filename: baseConfig.connection };
+  }
+
+  const config: Knex.Config = mergeDatabaseConfig(
+    baseConfig,
     {
       useNullAsDefault: true,
     },
     overrides,
   );
+
+  // If we don't create an in-memory database, interpret the connection string
+  // as a directory that contains multiple sqlite files based on the database
+  // name.
+  if (config.connection && typeof config.connection === 'object') {
```

Comment:
could this type be changed by the merge? we make it an object upfront but then check here again if i read i correctly

Reply:
It could be changed if the merge has a string as a `connection` field. This is probably never the case, as `contention{database: "xyz"}` is passed as an override.